### PR TITLE
Fix VS Cmake config for Png2Ico.

### PR DESCRIPTION
**Pull request by allexzander**
Signed-off-by: allexzander <blackslayer4@gmail.com>

### CODE REVIEW

File: cmake/modules/ECMAddAppIcon.cmake [hunk 1]
```diff
@@ -197,11 +197,15 @@ function(ecm_add_app_icon appsources)
 
         function(create_windows_icon_and_rc command args deps)
                 add_custom_command(
-                    OUTPUT "${_outfilename}.ico"
+                    OUTPUT "${_outfilebasename}.ico"
                     COMMAND ${command}
                     ARGS ${args}
                     DEPENDS ${deps}
                     WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+                    # the problem is Png2Ico tool does not understand .ico output path that contains spaces(even though the path is put into quotes")
```
Comment (er-vin) [inline cmake/modules/ECMAddAppIcon.cmake:205]: Just wondering if that's really the case or if it's simply that the OUTPUT needs further quoting? You know I'm thinking about tricks like `"\"${_outfilename}.ico\""`. Did you try this or similar?

Comment (allexzander) [reply cmake/modules/ECMAddAppIcon.cmake:205]: @er-vin I tried copying the command that gets executed during the build but fails with the "not expected at this time" error and it is complaining to a part of a path that goes after space and the quotes are there. So, it definitely doesn't like the space even in quotes.

Comment (allexzander) [reply cmake/modules/ECMAddAppIcon.cmake:205]: @er-vin I tried your suggestion. Here is what I am getting:

`FAILED: src/gui/Nextcloud.ico 
  cmd.exe /C "cd /D D:\work\nextcloud\desktop_vs\desktop\src\gui && "C:\Program Files\png2ico\png2ico.exe" "D:/work/nextcloud/desktop_vs/desktop/out/build/x64-Debug (default)/src/gui/Nextcloud.ico" D:/work/nextcloud/desktop_vs/desktop/theme/colored/16-Nextcloud-icon.png D:/work/nextcloud/desktop_vs/desktop/theme/colored/24-Nextcloud-icon.png D:/work/nextcloud/desktop_vs/desktop/theme/colored/32-Nextcloud-icon.png D:/work/nextcloud/desktop_vs/desktop/theme/colored/48-Nextcloud-icon.png D:/work/nextcloud/desktop_vs/desktop/theme/colored/64-Nextcloud-icon.png D:/work/nextcloud/desktop_vs/desktop/theme/colored/128-Nextcloud-icon.png D:/work/nextcloud/desktop_vs/desktop/theme/colored/256-Nextcloud-icon.png D:/work/nextcloud/desktop_vs/desktop/theme/colored/512-Nextcloud-icon.png D:/work/nextcloud/desktop_vs/desktop/theme/colored/1024-Nextcloud-icon.png"
  (default)/src/gui/Nextcloud.ico"" was unexpected at this time.`

**Comment by FlexW** [comment]
@er-vin @allexzander  It's a bug in the `png2.ico.exe`. I created a pr https://github.com/hiiamok/png2ImageMagickICO/pull/2. You can find my branch here https://github.com/FlexW/png2ImageMagickICO/tree/bugfix/quote-arg. I also created a binary from my branch here: https://cloud.nextcloud.com/s/dCBX6Nq3tf6NE96. If you use that executable the icon gets generated fine with no CMake changes. I think it's better to use the fixed executable than introducing hacks in our CMake scripts :)

**Comment by er-vin** [comment]
> I think it's better to use the fixed executable than introducing hacks in our CMake scripts :)

Yes definitely. That one should be closed then I guess.

**Comment by FlexW** [comment]
@allexzander I close this pr. Hope it is ok. Feel free to open it again if you think it is needed :)

**Pull request closed without merging**